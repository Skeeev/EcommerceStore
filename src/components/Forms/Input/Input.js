import React, { memo, useMemo } from 'react';
import { View, Text, TextInput } from 'react-native';
import isArray from 'lodash/isArray';
import isEqual from 'lodash/isEqual';

import { colors } from 'src/assets/styles/colors';
import styles from './Input.styles';

export const Input = memo(
  ({ icon, placeholderTextColor = colors.black, style, ...props }) => {
    const inputStyle = useMemo(
      () => (icon ? [...(isArray(style) ? style : [style]), styles.input] : style),
      [icon, style]
    );

    return (
      <View style={styles.field}>
        <Text style={styles.icon}>{icon}</Text>
        <TextInput {...props} style={inputStyle} placeholderTextColor={placeholderTextColor} />
      </View>
    );
  },
  ({ value: previousValue }, { value }) => isEqual(previousValue, value)
);

Input.displayName = 'Input';
