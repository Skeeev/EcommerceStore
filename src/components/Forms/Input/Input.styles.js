import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  field: {
    position: 'relative',
    justifyContent: 'center'
  },
  icon: {
    position: 'absolute',
    zIndex: 1,
    left: 30
  },
  input: {
    paddingRight: 15,
    paddingLeft: 70
  }
});
