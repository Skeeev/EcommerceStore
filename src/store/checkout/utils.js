import get from 'lodash/get';
import find from 'lodash/find';

const normalizeProducts = products =>
  products.map(({ key, name: title, price, total, quantity, thumb }) => ({
    id: `${key}`,
    title,
    price,
    total,
    quantity,
    imageUri: `http:${thumb}`
  }));

export const extractCheckoutData = ({ products, totals }) => ({
  products: normalizeProducts(products),
  total: get(find(totals, { id: 'total' }), 'text', '$00.00'),
  totalQuantity: products.reduce((total, { quantity }) => total + quantity, 0)
});
