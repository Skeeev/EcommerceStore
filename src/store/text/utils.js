export const prepareTextPayload = text => ({
  payload: {
    text
  }
});
