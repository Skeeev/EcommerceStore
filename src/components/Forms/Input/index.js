import { compose } from 'redux';
import { connect } from 'formik';

import { withFormikField } from 'src/HOCs/withFormikField';
import { Input } from './Input';

export * from './Input';
export default compose(connect, withFormikField)(Input);
