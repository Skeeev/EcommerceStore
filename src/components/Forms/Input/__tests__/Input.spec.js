import React from 'react';
import { shallow } from 'enzyme';

import { getComponentDisplayName } from 'src/utils/displayName';
import { Input } from '../Input';

describe(`${getComponentDisplayName(Input)} component`, () => {
  const renderComponent = props => shallow(<Input {...props} />);

  it('renders correctly', () => {
    const component = renderComponent({
      name: 'dummy-name',
      value: 'dummy value',
      placeholder: 'dummy placeholder',
      onChangeText: jest.fn()
    });

    expect(component).toMatchSnapshot();
  });
});
