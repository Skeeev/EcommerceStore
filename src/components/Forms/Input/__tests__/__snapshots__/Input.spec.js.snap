// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders correctly 1`] = `
<View
  style={
    Object {
      "justifyContent": "center",
      "position": "relative",
    }
  }
>
  <Text
    style={
      Object {
        "left": 30,
        "position": "absolute",
        "zIndex": 1,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    name="dummy-name"
    onChangeText={[MockFunction]}
    placeholder="dummy placeholder"
    placeholderTextColor="#000000"
    rejectResponderTermination={true}
    underlineColorAndroid="transparent"
    value="dummy value"
  />
</View>
`;
